File: tools/binarySearch.js
const binarySearch = (arr, target) => {
  if (!arr) return 0;
  arr.sort((a, b) => a - b);
  let start = 0;
  let end = arr.length - 1;
  while (start < end) {
    let mid = Math.floor((end + start) / 2);
    if (arr[mid] >= target) end = mid;
    else start = mid + 1;
  }
  return arr.length - start;
};

File: goorm/BinarySearch.py
# 2021.12.20. goorm level
# Binary Search


def binary_search(N, li, target):
    start = 0
    end = N-1
    while start < end:
        mid = (start+end)//2
        if li[mid] == target:
            return mid+1
        elif li[mid] < target:
            start = mid+1
        else:
            end = mid-1
    return 'X'


N = int(input())
num_list = list(map(int, input().split()))
target = int(input())

print(binary_search(N, num_list, target))
